sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -104,95 +84,57 @@ import {
+  "#Microsoft.Skills.Util.ShaperSkill": true,
+  "#Microsoft.Skills.Vision.ImageAnalysisSkill": true,
+  "#Microsoft.Skills.Vision.OcrSkill": true,
+  "#Microsoft.Skills.Custom.AmlSkill": true,
+  "#Microsoft.Skills.Vision.VectorizeSkill": true,
+};
 
 export function convertSkillsToPublic(skills: SearchIndexerSkillUnion[]): SearchIndexerSkill[] {
   if (!skills) {
     return skills;
   }
 
-  const result: SearchIndexerSkill[] = [];
-  for (const skill of skills) {
-    switch (skill.odatatype) {
-      case "#Microsoft.Skills.Util.ConditionalSkill":
-        result.push(skill as ConditionalSkill);
-        break;
-      case "#Microsoft.Skills.Text.KeyPhraseExtractionSkill":
-        result.push(skill as KeyPhraseExtractionSkill);
-        break;
-      case "#Microsoft.Skills.Vision.OcrSkill":
-        result.push(skill as OcrSkill);
-        break;
-      case "#Microsoft.Skills.Vision.ImageAnalysisSkill":
-        result.push(skill as ImageAnalysisSkill);
-        break;
-      case "#Microsoft.Skills.Text.LanguageDetectionSkill":
-        result.push(skill as LanguageDetectionSkill);
-        break;
-      case "#Microsoft.Skills.Util.ShaperSkill":
-        result.push(skill as ShaperSkill);
-        break;
-      case "#Microsoft.Skills.Text.MergeSkill":
-        result.push(skill as MergeSkill);
-        break;
-      case "#Microsoft.Skills.Text.EntityRecognitionSkill":
-        result.push(skill as EntityRecognitionSkill);
-        break;
-      case "#Microsoft.Skills.Text.SentimentSkill":
-        result.push(skill as SentimentSkill);
-        break;
-      case "#Microsoft.Skills.Text.SplitSkill":
-        result.push(skill as SplitSkill);
-        break;
-      case "#Microsoft.Skills.Text.PIIDetectionSkill":
-        result.push(skill as PIIDetectionSkill);
-        break;
-      case "#Microsoft.Skills.Text.V3.EntityRecognitionSkill":
-        result.push(skill as EntityRecognitionSkillV3);
-        break;
-      case "#Microsoft.Skills.Text.V3.EntityLinkingSkill":
-        result.push(skill as EntityLinkingSkill);
-        break;
-      case "#Microsoft.Skills.Text.V3.SentimentSkill":
-        result.push(skill as SentimentSkillV3);
-        break;
-      case "#Microsoft.Skills.Text.TranslationSkill":
-        result.push(skill as TextTranslationSkill);
-        break;
-      case "#Microsoft.Skills.Custom.WebApiSkill":
-        result.push(skill as WebApiSkill);
-        break;
-      case "#Microsoft.Skills.Text.CustomEntityLookupSkill":
-        result.push(skill as CustomEntityLookupSkill);
-        break;
-      case "#Microsoft.Skills.Util.DocumentExtractionSkill":
-        result.push(skill as DocumentExtractionSkill);
-        break;
-      case "#Microsoft.Skills.Custom.AmlSkill":
-        result.push(skill as AzureMachineLearningSkill);
-        break;
-    }
-  }
-  return result;
+  // This validation has already GAed
+  return skills.filter((skill): skill is SearchIndexerSkill => knownSkills[skill.odatatype]);
```

Comment:
Consider using a `Set` instead of a map with boolean values here, and then you'll just use `knownSkills.has`

Reply:
I used `Record` here because it requires that every field is present in the object literal. Is there a way to use `Set` that would still cause tsc to catch a forgotten skill?